your-toolbelt.md

```diff
@@ -0,0 +1,47 @@
+### Your toolbelt
+
+
+#### Exercise: Uses and understands unit, functional and integration tests
+
+1. You've now written two types of tests. Name them, and explain the differences.
+The test's writen were the Unit Tests and the Integration Tests. The Unit Tests are to teste a small piece of code, mainly to test the models (the views and the controllers aren't so much tested, in case of the views because it can change in a short period of time) Ideally, there will be, at least, one test for each method. The Integration Tests are maded to test differentes pieces of code of the system and garantee the they work together. In the simple sinatra sitiuation, the integration tests were to navegate in the routes, fill a specific field, click a button and expect the page to have a specific text.
+
+2. The other tests we haven't made are the Functional Tests, that are the tests to the controllers. The controllers receive incomming web requeste and eventually respond with a view.
+A simple case will be 
+
+```ruby
+it 'should be testable' do
+  get '/'
+  last_response.should be_ok
+end
+``
+#### Exercise: Knows how Capybara and drivers work
+
+1. What type of tests do we use Capybara for? What are its strengths and limitations?
+-- We use the capybara to do integration tests, by simulating how a real user interacts with the app.
+The advantages of using Capybara are:
+a. No setup necessary for rails and rack applications. Works out of the box.
+b. Intuitiva API 
+c. Switch the backend - The tests uses a simulated engine of the browser, without the overhead.
+d. Powerful synchronization - so that we never have to manually wait for the asynchronous processes to complete. 
```

Comment:
This is talking about situations such as:

```
click_on "Link that goes to other page"
expect(page).to have_text "Some text"
```

Capybara will take care of waiting for the page that loads when you click the link, to finish loading and only run the expectation then.